modules/Bio/EnsEMBL/Variation/VariationFeature.pm

```diff
@@ -1968,7 +1967,84 @@ sub hgvs_genomic {
 
 }
 
+=head2  spdi_genomic
+
+  Description : Returns a reference to a hash with the allele as key and a string with the genomic SPDI notation of this VariationFeature as value. By default uses the
+                slice it is placed on as reference. 
+  Returntype  : Hash reference
+  Exceptions  : Throws exception if VariationFeature can not be described relative to the feature_Slice of the supplied reference feature
+  Caller      : general
+  Status      : Experimental
+
+=cut
+
+sub spdi_genomic{ 
+
+  my $self = shift; 
+  
+  my $ref_feature = $self;   
+  my %spdi; 
+
+  my $ref_slice = $self->slice; 
+ 
+  my ($vf_start, $vf_end, $ref_length) = ($ref_feature->start, $ref_feature->end, ($ref_feature->end - $ref_feature->start) + 1);
+
+  return {} if ($vf_start < 1 || $vf_end < 1);  
+
+  my $vf_strand = $self->strand(); 
+  
+  # set up sequence reference 
+  my $syn = $ref_slice->get_all_synonyms('RefSeq_genomic'); 
+  my $reference_name = (defined $syn->[0] ? $syn->[0]->name() : $ref_feature->seq_region_name()); 
+
+  my @all_alleles = split(/\//,$ref_feature->allele_string());
+  my $ref_allele = shift @all_alleles; 
+  my $spdi_ref_allele; 
+  my $spdi_alt_allele; 
+  
+  # Create a spdi notation for each allele 
+  foreach my $alt_allele (@all_alleles){ 
```

Comment:
I know it wasn't in the original spec, but it would be useful to optionally return the reference spdi. We could have a boolean (include_reference_allele?) optionally passed in to turn this on and it could default to 0 ( alt alleles only) without.